core/src/main/java/org/apache/calcite/rel/rules/ProjectCalcMergeRule.java

```diff
@@ -82,8 +82,6 @@ public ProjectCalcMergeRule(RelBuilderFactory relBuilderFactory) {
             project.getRowType(),
             cluster.getRexBuilder());
     if (RexOver.containsOver(program)) {
-      LogicalCalc projectAsCalc = LogicalCalc.create(calc, program);
```

Comment:
Isn't this a breaking change? Currently the Project is removed, with this proposal the Project remains.

Reply:
Unfortunately, yes. But i assume that special rule `ProjectToCalcRule` can perform simple conversion from `LogicalProject` to `LogicalCalc`. Whereas `ProjectCalcMergeRule` will be able to stay more clear and more generalize.

Reply:
This is a small breaking change. I doubt people will notice but let's put a small note in `history.md` to be on the safe side. Moreover, please remove the now obsolete comments in lines 73-75 above.